lambda-runtime/src/AWS/Lambda/Runtime/Client.hs

```diff
@@ -63,20 +63,20 @@ getConnection = do
         }
   pure $ Connection{..}
 
-getNextEvent:: Connection -> LambdaClient Event
+getNextEvent:: JSON.FromJSON a => Connection -> LambdaClient (Event a)
 getNextEvent connection = do
   response <- getNextEventValue connection
 
-  requestId <- RequestId <$> liftEither (fetchHeader response "Lambda-Runtime-Aws-Request-Id")
-  traceId   <- parseTraceHeader =<< liftEither (fetchHeader response "Lambda-Runtime-Trace-Id")
+  requestId   <- RequestId <$> liftEither (fetchHeader response "Lambda-Runtime-Aws-Request-Id")
+  traceHeader <- parseTraceHeader =<< liftEither (fetchHeader response "Lambda-Runtime-Trace-Id")
```

Comment:
This is really a trace header not a trace id, trace headers contain parent segments etc which this value does. Hence naming it accordingly.